cni/telemetry/client/telemetry_client.go

```diff
@@ -0,0 +1,72 @@
+package telemetryclient
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/Azure/azure-container-networking/telemetry"
+	"go.uber.org/zap"
+)
+
+const (
+	telemetryNumberRetries          = 5
+	telemetryWaitTimeInMilliseconds = 200
+)
+
+type TelemetryClient struct {
+	CNIReportSettings *telemetry.CNIReport
+	tb                *telemetry.TelemetryBuffer
+	logger            *zap.Logger
+	lock              sync.Mutex
+}
+
+var Telemetry = NewTelemetryClient(&telemetry.CNIReport{})
+
+func NewTelemetryClient(report *telemetry.CNIReport) *TelemetryClient {
+	return &TelemetryClient{
+		CNIReportSettings: report,
+	}
+}
+
+func (c *TelemetryClient) ConnectTelemetry(logger *zap.Logger) {
```

Comment:
why can't this be part of the constructor? wouldn't that remove the need for nil checks?

Reply:
If we `ConnectTelemetry` in `NewTelemetryClient` I believe we would try to connect to the telemetry service during tests, which I do not want to occur. `Connect / StartAndConnect` are also called by different main methods (stateless vs. non-stateless cni) whereas `NewTelemetryClient` is called for both. Adding attempts to _only_ connect to `NewTelemetryClient` (via `ConnectTelemetry`) would be incorrect behavior in non-stateless cni because we want to start the telemetry service before any connection is attempted.